src/NHSD.BuyingCatalogue.Solutions.API/Controllers/Hostings/OnPremiseHostingController.cs

```diff
@@ -0,0 +1,36 @@
+using System.ComponentModel.DataAnnotations;
+using System.Net;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using NHSD.BuyingCatalogue.Solutions.API.ViewModels.Hostings;
+using NHSD.BuyingCatalogue.Solutions.Contracts.Queries;
+
+namespace NHSD.BuyingCatalogue.Solutions.API.Controllers.Hostings
+{
+    [Route("api/v1/solutions")]
+    [ApiController]
+    [Produces("application/json")]
+    [AllowAnonymous]
+    public sealed class OnPremiseHostingController : ControllerBase
+    {
+        private readonly IMediator _mediator;
+
+        public OnPremiseHostingController(IMediator mediator) =>
+            _mediator = mediator;
+
+        [HttpGet]
+        [Route("{id}/sections/hosting-type-on-premise")]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType((int)HttpStatusCode.NoContent)]
+        [ProducesResponseType((int)HttpStatusCode.NotFound)]
+        public async Task<ActionResult> GetPremiseHosting([FromRoute] [Required] string id)
```

Comment:
OnPremise?

I would just call them Get and Update though, we know they're for OnPremise as that's what the controller is